composer.json

```diff
@@ -17,36 +17,37 @@
         }
     ],
     "require": {
-        "symfony/http-foundation": "~2.3|~3.0",
-        "symfony/form": "^2.3.5|~3.0",
-        "symfony/validator": "~2.3|~3.0",
-        "symfony/security-bundle": "~2.3|~3.0",
-        "symfony/routing": "~2.3|~3.0",
-        "symfony/config": "^2.3.9|~3.0",
-        "symfony/console": "~2.3|~3.0",
-        "symfony/twig-bridge": "^2.3.5|~3.0",
-        "symfony/class-loader":"~2.3|~3.0",
-        "symfony/expression-language": "~2.4|~3.0",
-        "symfony/templating":"~2.3|~3.0",
-        "symfony/translation":"~2.3|~3.0",
-        "symfony/dependency-injection": "^2.3.3|~3.0",
-        "symfony/property-access": "~2.3|~3.0",
-        "symfony/security-acl": "~2.3|~3.0",
-        "twig/twig": "~1.23",
-        "twig/extensions": "~1.0",
-        "sonata-project/exporter": "~1.0",
-        "sonata-project/block-bundle": "^2.3.9",
-        "sonata-project/core-bundle": "^2.3.10",
-        "doctrine/common": "~2.2",
-        "doctrine/inflector": "~1.0",
+        "php": "^5.3 || ^7.0",
+        "symfony/http-foundation": "^2.3 || ^3.0",
+        "symfony/form": "^2.3.5 || ^3.0",
+        "symfony/validator": "^2.3 || ^3.0",
+        "symfony/security-bundle": "^2.3 || ^3.0",
+        "symfony/routing": "^2.3 || ^3.0",
+        "symfony/config": "^2.3.9 || ^3.0",
+        "symfony/console": "^2.3 || ^3.0",
+        "symfony/twig-bridge": "^2.3.5 || ^3.0",
+        "symfony/class-loader":"^2.3 || ^3.0",
+        "symfony/expression-language": "^2.4 || ^3.0",
+        "symfony/templating":"^2.3 || ^3.0",
+        "symfony/translation":"^2.3 || ^3.0",
+        "symfony/dependency-injection": "^2.3.3 || ^3.0",
+        "symfony/property-access": "^2.3 || ^3.0",
+        "symfony/security-acl": "^2.3 || ^3.0",
+        "twig/twig": "^1.23",
+        "twig/extensions": "^1.0",
+        "sonata-project/exporter": "^1.0",
+        "sonata-project/block-bundle": "3.x-dev@dev",
+        "sonata-project/core-bundle": "3.x-dev@dev",
+        "doctrine/common": "^2.2",
+        "doctrine/inflector": "^1.0",
         "knplabs/knp-menu-bundle": "^2.1.1"
     },
     "require-dev": {
-        "jms/di-extra-bundle": "~1.7",
-        "sensio/generator-bundle": "~2.3|~3.0",
-        "symfony/yaml": "~2.3|~3.0",
+        "jms/di-extra-bundle": "^1.7",
+        "sensio/generator-bundle": "^2.3 || ^3.0",
+        "symfony/yaml": "^2.3 || ^3.0",
```

Comment:
It sorts all package, but this works with `composer require` only. https://getcomposer.org/doc/06-config.md#sort-packages